src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -231,11 +231,42 @@ private void SetStateForCustomizationTask(TaskJSONToCSClasses.BaseClass response
 
                     if (_oldUnitState[i] != responseStatus)
                     {
-                        var task = _units[i];
-                        var unitState = DevBoxOperationHelper.JSONStatusToUnitStatus(responseStatus);
-                        var resultInfo = responseStatus == "Failed" ? _commonfailureResult : null;
-                        ConfigurationSetStateChangedEventArgs args = new(new(ConfigurationSetChangeEventType.UnitStateChanged, setState, unitState, resultInfo, task));
-                        ConfigurationSetStateChanged?.Invoke(this, args);
+                        var task = _units[i];
+                        if (responseStatus == "Failed")
+                        {
+                            // Wait for a few seconds before fetching the logs
+                            // otherwise the logs might not be available
+                            Thread.Sleep(TimeSpan.FromSeconds(15));
+
+                            // Make the log API string : Remove the API version from the URI and add 'logs'
+                            var logURI = _restAPI.Substring(0, _restAPI.LastIndexOf('?'));
+                            var id = response.Tasks[i].Id;
+                            logURI += $"/logs/{id}?{Constants.APIVersion}";
+
+                            // Fetch the logs
+                            var log = _managementService.HttpsRequestToDataPlaneRawResponse(new Uri(logURI), _devId, HttpMethod.Get).GetAwaiter().GetResult();
+
+                            // Split the log into lines
+                            var logLines = log.Split('\n');
+
+                            // Find index of line with "Result" in it
+                            // The error message is in the next line
+                            var errorIndex = Array.FindIndex(logLines, x => x.Contains("Result")) + 1;
+                            var errorMessage = errorIndex > 0 ? logLines[errorIndex] : Resources.GetResource(Constants.DevBoxCheckLogsKey);

```

Comment:
Could logLines ever be empty?

Reply:
I don't think so. If there is a failure, there will be a log file. And if there isn't, the API will throw and that will be caught.
Do you think it can be empty in some other case?

Reply:
There are other questions here: can logLines be without "Result" and can "Result" be in the first line?
Array.FindIndex returns -1 if value is not found and since it's a zero-based index the first line is at logLines[0].

Even if "Result" is not expected to be in the first line I think checking against -1 (or >= 0) is more correct.

var errorMessage = errorIndex >= 0 ? logLines[errorIndex] : Resources.GetResource(Constants.DevBoxCheckLogsKey);
```suggestion
                            var errorMessage = errorIndex >= 0 ? logLines[errorIndex] : Resources.GetResource(Constants.DevBoxCheckLogsKey);

```

Reply:
Yup, there can be no "Result". It is accounted for.